docs/DeveloperGuide.md

```diff
@@ -1,38 +1,433 @@
+  - [Appendix D: Glossary](#appendix-d-glossary)
+  - [Appendix E: Acknowledgements](#appendix-e-acknowledgements)
+    - [Third-party libraries](#third-party-libraries)
 
-## Design & implementation
+[Appendix A: Product Scope](#appendix-a-product-scope) <br>
+[Appendix B: User Stores](#appendix-b-user-stories) <br>
+[Appendix C: Non-Functional Requirements](#appendix-c-non-functional-requirements) <br>
+[Appendix D: Glossary](#appendix-d-glossary) <br>
+[Appendix E: Acknowledgements](#appendix-e-acknowledgements) <br>
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 1. Introduction
 
+### 1.1. Welcome!
+
+Welcome, and thank you for choosing to help contribute to Yet Another Module Organiser/ Manager!
+Yet Another Module Organizer and Manager (YAMOM) is an all-in-one desktop app featuring
+a full course catalogue, module search and timetable builder for the National University of Singapore, optimized for use via a Command Line Interface.
+
+This document intends to onboard developers onto YAMOM. We hope to bring you in to fix bugs, or even adding
+new features and dimensions to YAMOM!
+It gives insights on how the project is set up, the architecture used,
+and the code style one should adopt when contributing to the project.
+
+## 2. Setting up
+
+This section describes the development tools used in the creation of YAMOM.
+
+### 2.1. Prerequisites
+
+1. **JDK** 11
+2. **IntelliJ** IDEA
+3. **Gradle** 6.2
+
+### 2.2. Setting up the project locally
+
+1. **Fork** this repo, and **clone** the fork into your computer.
+2. Open IntelliJ (if you are not in the welcome screen, click **`File`** > **`Close Project`** to close the existing project dialog first).
+3. Set up the correct JDK version for Gradle  
+   a.To set up the correct project structure **`Configure`** > **`Project Defaults`** > **`Project Structure`**  
+   b. Under **`New...`** find the directory of the appropriate JDK version.
+4. Click **`Import Project`**.
+5. Find the **`build.gradle`** file and select it. Click **`OK`**.
+6. Click **`Open as Project`**.
+7. Click **`OK`** and accept the default settings.
+
+### 2.3. Configure the coding style
+
+In IntelliJ's IDEA we adopt [[se-edu/guides] IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html)
+to set up IDEA’s coding style to match ours.
+
+> Optionally, you can follow the guide [[se-edu/guides] Using Checkstyle](https://se-education.org/guides/tutorials/checkstyle.html)
+> to find how to use the CheckStyle within IDEA e.g., to report problems as you write code.
+
+## 3. Design
+
+<!-- {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.} -->
+
+### 3.1 Architecture
+
+**How the architecture components interact with each other**
+
+Core program flow is managed by the Duke class.
+
+![Main Program Flow](images/mainProgramFlow.png)
```

Comment:
<img width="433" alt="pr1" src="https://user-images.githubusercontent.com/116053971/198247409-ae56096e-67d0-49b2-b913-c79175966684.png">

Do remember to deactivate the activation bar so that there is closure.